### DISCUSSION TITLE
IP Pools, ip rotation on sending an existing feature?

### DISCUSSION THREAD

**Discussion by alex-pommier** [postalserver/postal#2352] [Development discussions]
Either my IP pools ip rotation is not working or it's not an existing feature. I thought that by configuring IP Pools, postal would rotate all IPs in the pool if set to the same priority. I've noticed that ALL emails are being sent only from 1 IP in my pool. Checking on google postmaster tools my volume became quite high and only 1 IP has data of my 4 IPs configured. When I switched the priority of the 2nd IP in my pool to a higher priority (the no data IP) emails started going through. I sent some tests to my email server 50 emails and all arrived from the same IP when switching back the pool ip priority for all 4 IPs to 25. 

Is the normal behavior of postal's IP pools to auto-rotate between all IPs in the pool unless a given IP has a higher priority? If so that does not seem to be what's happening.

Ideally if I have X IPs configured and all are set to the same priority level, postal would rotate between ALL IPs in the pool to maintain a good sender reputation.

**Comment by willpower232** [comment]
Have a look at #2209, I'm still not a mathematician but apparently it is not as obvious as you might hope.

**Reply by alex-pommier** [reply]
Hi Will, I read through that post thank you for linking it. Still though on there they describe the ratio's difference for me I don't really care if it's an exact number but that the sending goes through a somewhat balanced rotation between IPs in my pool. If the numbers are off a bit that's fine. In my case from what i've seen ALL emails are being sent from only 1 IP with pools properly configured and all priorities set equally across the IP pool. I've not tested by setting IE: all 4 of my IPs to 75 to see what happens that seemed counter-intuitive but I can try it. Let me know your thoughts.
